staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -770,10 +797,19 @@ func (c *Cacher) GetList(ctx context.Context, key string, opts storage.ListOptio
 	if err != nil {
 		return err
 	}
-	if listRV == 0 && !c.ready.check() {
-		// If Cacher is not yet initialized and we don't require any specific
-		// minimal resource version, simply forward the request to storage.
-		return c.storage.GetList(ctx, key, opts, listObj)
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.ResilientWatchCacheInitialization) {
+		if !c.ready.check() && shouldDelegateListOnNotReadyCache(opts) {
+			// If Cacher is not initialized, delegate List requests to storage
+			// as described in https://kep.k8s.io/4568
+			return c.storage.GetList(ctx, key, opts, listObj)
```

Comment:
have we considered only allowing the local client to get through and rejecting all other requests?

Reply:
I wanted to start simple and introduce the mechanism itself first.

We can iterate on exact conditions as a follow up [at this level we don't really have this information so it would require plumbing that through multiple layers]